### ISSUE TITLE
Nullness checker incorrect error message: "found: @UnderInitialization(java.lang.Object.class)"  for method annotated as "@UnknownInitialization"

### ISSUE THREAD

**Issue by tanejasomi** [typetools/checker-framework#1879]
Hi, Initialization checker issue error "required @UnknownInitialization annotation" for method already annotated as @UnknownInitialization. Please find below sample code executed on checker framework live demo site.
```
import org.checkerframework.checker.nullness.qual.*;
import org.checkerframework.checker.initialization.qual.*;
class Test {
    String val;
    String type;
    Test(){
        this.getType();
        }
        
    @EnsuresNonNull({"type","val"})    
    void getType(@UnknownInitialization(Test.class) Test this){
        this.type = CalculateType();
            }
            
    @EnsuresNonNull("val")
    String CalculateType(@UnknownInitialization(Test.class) Test this){
        //Some processing
         this.val = "abc";
         type ="xyz";
         return type;
        }
}
```
Expected Output: No error (or some other warning)
Actual Output: 
```
Error: [method.invocation.invalid] call to getType() not allowed on the given receiver.
   found   : @UnderInitialization(java.lang.Object.class) @NonNull Test
   required: @UnknownInitialization(Test.class) @NonNull Test
```

When method getType is already annotated with @UnknownInitialization checker framework consider it as @UnderInitialization.
My understanding for issue is may be when analyzing method call in constructor it sets default value "@UnderInitialization" and does not set again as @UnknownInitialization when checking signature of method. This understanding could be incorrect maybe its just case of wrong error message.

**Comment by wmdietl** [comment]
This is the correct behavior.
Note how class `Test` has non-null fields `val` and `type`.
At the entry of the `Test()` constructor, only the `Object` constructor has been executed.
So the type of `this` in `this.getType()` is `@UnderInitialization(Object.class)`: we definitely know that we're initializing the object, b/c we're still in the constructor. So far we only initialized the `Object` part, so that is the frame.
That explains the `found` type in the method invocation.

The `expected` type is the declared receiver of method `getType`, which is `@UnknownInitialization(Test.class)`. This type means: we don't know or care whether the object is being initialized (unknown), but we do know that at least the `Test` fields have been initialized.

Now you should see the mismatch in the assumptions between these two: the actual receiver doesn't guarantee that the fields of class `Test` are initialized, which the method requires.

Please let us know if this can be clarified in https://checkerframework.org/manual/#initialization-checker
